components/brave_wallet/browser/eth_tx_controller.cc

```diff
@@ -429,6 +440,8 @@ void EthTxController::ProcessLedgerSignature(
     std::move(callback).Run(false);
     return;
   }
+  meta->status = mojom::TransactionStatus::Approved;
+  tx_state_manager_->AddOrUpdateTx(*meta);
```

Comment:
could there not be more than one transaction in progress at a time? Maybe in more than one tab/window?

Reply:
I guess this is a general issue and not specific to this PR, but this seems dangerous to me